File: ToDoList2/app/src/main/java/com/mike/todolist/data/model/Task.java
package com.mike.todolist.data.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.annotation.NonNull;

import java.io.Serializable;
import java.util.Date;

@Entity(tableName = "tasks")
public class Task implements Serializable {

    @PrimaryKey(autoGenerate = true)
    private int id;

    @NonNull
    private String title;

    private Date dueDate;

    @NonNull
    private String priority; // "Alta", "Media", "Baja"

    private boolean isCompleted;

    private boolean isScheduled;

    // Constructor
    public Task(@NonNull String title, Date dueDate, @NonNull String priority, boolean isCompleted, boolean isScheduled) {
        this.title = title;
        this.dueDate = dueDate;
        this.priority = priority;
        this.isCompleted = isCompleted;
        this.isScheduled = isScheduled;
    }

    // Getters y Setters

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @NonNull
    public String getTitle() {
        return title;
    }

    public void setTitle(@NonNull String title) {
        this.title = title;
    }

    public Date getDueDate() {
        return dueDate;
    }

    public void setDueDate(Date dueDate) {
        this.dueDate = dueDate;
    }

    @NonNull
    public String getPriority() {
        return priority;
    }

    public void setPriority(@NonNull String priority) {
        this.priority = priority;
    }

    public boolean isCompleted() {
        return isCompleted;
    }

    public void setCompleted(boolean completed) {
        isCompleted = completed;
    }

    public boolean isScheduled() {
        return isScheduled;
    }

    public void setScheduled(boolean scheduled) {
        isScheduled = scheduled;
    }
}

File: ToDoList/Models/Task.swift
//
//  Task.swift
//  ToDoList
//
//  Created by Miguel Cuponerapp on 13/12/24.
//

import Foundation
import SwiftData

enum TaskPriority: String, Codable, CaseIterable {
    case alta = "Alta"
    case media = "Media"
    case baja = "Baja"
}

@Model
class Task: Identifiable {
    var id: UUID
    var title: String
    var dueDate: Date?
    var priority: TaskPriority
    var isCompleted: Bool
    var isScheduled: Bool

    init(
        title: String,
        dueDate: Date? = nil,
        priority: TaskPriority = .media,
        isCompleted: Bool = false,
        isScheduled: Bool = false
    ) {
        self.id = UUID()
        self.title = title
        self.dueDate = dueDate
        self.priority = priority
        self.isCompleted = isCompleted
        self.isScheduled = isScheduled
    }
}

extension Task {
    var isOverdue: Bool {
        if let dueDate = dueDate {
            return dueDate < Date() && !isCompleted
        }
        return false
    }
}
